### PR TITLE
sketch of a session implementation

### PR DESCRIPTION
**Pull request by wenzowski**
The `session` field will only be available when authenticated via
keycloak. In order to use the `userBySub` relation, a `User` must
already exist with the `user.uuid` column set to the `session.sub`.

example usage:
```graphql
{
  session {
    jti
    exp
    nbf
    iat
    iss
    aud
    sub
    typ
    azp
    authTime
    sessionState
    acr
    emailVerified
    name
    preferredUsername
    givenName
    familyName
    email
    userBySub {
      id
      rowId
      uuid
      firstName
      lastName
      emailAddress
      createdAt
      updatedAt
      createdBy
      updatedBy
      userOrganisationsByUserId {
        edges {
          node {
            id
          }
        }
      }
    }
  }
}
```

Comments describing the `JwtToken` type conformance to [RFC 7519] are
still needed, and the nullability of those fields should be correctly
set in the type to conform to the RFC.

[RFC 7519]: https://tools.ietf.org/html/rfc7519

### CODE REVIEW

**Comment by wenzowski** [comment]
tests for this blocked on #86

**Comment by wenzowski** [comment]
rebased on develop

**Comment by wenzowski** [comment]
rebased on develop

File: schema/deploy/table_user.sql [hunk 1]
```diff
@@ -48,7 +48,7 @@ comment on column ggircs_portal.user.updated_by is 'The person who updated the u
 insert into ggircs_portal.user (id, uuid, first_name, last_name, email_address, occupation, phone_number)
         values (1, '6c01258f-6ad8-4790-8ccc-485163f122a5' , 'Hamza', 'Javed', 'hamza@button.is', 'developer', '6043345678');
 insert into ggircs_portal.user (id, uuid, first_name, last_name, email_address, occupation, phone_number)
-        values (2, 'ca716545-a8d3-4034-819c-5e45b0e775c9' , 'Alec', 'Wenzowski', 'alec@button.is', 'CEO', '7789906789');
+        values (2, 'ca716545-a8d3-4034-819c-5e45b0e775c9' , 'Alec', 'Wenzowski', 'alec@button.is', 'crazy person', '7789906789');
```
Comment (matthieu-foucault) [inline schema/deploy/table_user.sql]: :upside_down_face:

File: schema/verify/type_jwt_token.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+-- Verify ggircs-portal:type_jwt_token on pg
+
+BEGIN;
+
+-- XXX Add verifications here.
+
+ROLLBACK;
```
Comment (matthieu-foucault) [inline schema/verify/type_jwt_token.sql]: no verify?

Comment (wenzowski) [reply schema/verify/type_jwt_token.sql]: implemented

File: schema/deploy/table_user.sql (hunk 1 above)
Comment (wenzowski) [reply schema/deploy/table_user.sql]: reverted

**Comment by wenzowski** [comment]
rebased on develop

**Comment by wenzowski** [comment]
rebased on develop

**Comment by wenzowski** [comment]
@matthieu-foucault do you want to merge this before #103? It fixes a lot of linter stuff in passing...

**Comment by matthieu-foucault** [comment]
Will rebase and merge

File: app/components/Layout/Header.js [hunk 3]
```diff
@@ -15,62 +16,52 @@ const Header = props => (
...
+                        Profile
+                      </Button>
+                    </Form>
+                  </Col>
+                  <Col>
+                    <Form action="/logout" method="post">
+                      <Button type="submit" variant="outline-light">
+                        Logout
+                      </Button>
+                    </Form>
```
Comment (wenzowski) [inline app/components/Layout/Header.js]: @matthieu-foucault this triggers a full re-render and the Location header is respected by the browser as a result

File: app/server/index.js [hunk 4]
```diff
@@ -64,25 +64,52 @@ app.prepare().then(() => {
...
+
+        return claims;
       }
     })
   );
 
-  server.get('/form-builder', keycloak.protect());
+  server.post('/login', keycloak.protect(), (req, res) =>
+    res.redirect(302, '/user-dashboard')
+  );
```
Comment (wenzowski) [inline app/server/index.js]: @matthieu-foucault the 302 redirect here assumes the error case is handled by a correctly configured `keycloak.protect()` middleware and that there's either a login form embedded on the page, or a discrete login page with a form configured to produce this post request

**Pull request merged**